src/fortrancode.l

```diff
@@ -732,6 +733,54 @@ LANGUAGE_BIND_SPEC BIND{BS}"("{BS}C{BS}(,{BS}NAME{BS}"="{BS}"\""(.*)"\""{BS})?")
 %%
  /*==================================================================*/
 
+ /*------ left-hand-side variable references/assignment  ------------*/
+ /*       catch these early so they don't get flagged as keywords.   */
+
+<Start>{ID}{BS}/[=%]                    {   // scalar assignment or type structure
+                                            //printf("===> line number %d lhs-variable |%s|\n",g_yyLineNr,yytext);
+  					    g_insideBody=TRUE;
+                                            // extract variable name and process it
+                                            char *tmp= (char *) malloc(strlen(yytext)+1);
+                                            strcpy(tmp,yytext);
+                                            strtok(tmp," \t(");
+                                            //printf("     generateLink |%s|\n",tmp);
+                                            generateLink(*g_code, tmp);
+                                            if ( strlen(tmp) != strlen(yytext) ) {
+                                              // everything else is thrown back into input stream
+                                              // to prevent {BS} from being gobbled
+                                              //printf("     reprocess |%s|\n",yytext+strlen(tmp));
+                                              yyless(strlen(tmp));
+                                            }
+                                            free(tmp);
+					    g_insideBody=FALSE;
+                                        }
+
```

Comment:
Why not put the `{BS}` part after the `/` so the push back is not necessary
Why in the strtok an check for `(`, cannot be this rule

Reply:
On line 739, {BS} after / would be nice, but generates "dangerous trailing context" compiler warning because {ID} contains a variable length wildcard (*).  Please suggest a simpler alternative that works.

On line 745, strtok is intended to capture {BS} and should NOT contain "(".  It is a remnant from a previous iteration that tried to capture array assignment within a common pattern match.  I will update.